app/models/do_not_contact.rb

```diff
@@ -0,0 +1,28 @@
+# Store details of people who have asked you not to contact them at all
+# Uses EmailAddress.redact to store/find addresses as a one-way hash
+class DoNotContact < ApplicationRecord
+  validates :email, presence: true, uniqueness: true
+
+  before_save :redact_email
+
+  def self.include?( email )
+    find_by( email: canonicalise_and_redact( email ) ) ? true : false
```

Comment:
This can be reduced using `exists?`:

```diff
-find_by( email: canonicalise_and_redact( email ) ) ? true : false
+exists?( email: canonicalise_and_redact( email ) )
```